environment/MetaData.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build integration
+// +build integration
+
+package environment
+
+import (
+	"flag"
+	"github.com/aws/amazon-cloudwatch-agent-test/environment/compute_type"
+	"github.com/aws/amazon-cloudwatch-agent-test/environment/ecs_deployment_type"
+	"github.com/aws/amazon-cloudwatch-agent-test/environment/ecs_launch_type"
+	"github.com/aws/amazon-cloudwatch-agent-test/test"
+	"log"
+)
+
+type MetaData struct {
+	ComputeType               compute_type.ComputeType
+	EcsLaunchType             ecs_launch_type.ECSLaunchType
+	EcsDeploymentStrategy     ecs_deployment_type.ECSDeploymentType
+	EcsClusterArn             string
+	EcsClusterName            string
+	CwagentConfigSsmParamName string
+	EcsServiceName            string
+}
+
+type MetaDataStrings struct {
+	ComputeType               string
+	EcsLaunchType             string
+	EcsDeploymentStrategy     string
+	EcsClusterArn             string
+	CwagentConfigSsmParamName string
+	EcsServiceName            string
+}
+
+func registerComputeType(dataString *MetaDataStrings) {
+	flag.StringVar(&(dataString.ComputeType), "computeType", "", "EC2/ECS/EKS")
+}
+func registerECSData(dataString *MetaDataStrings) {
+	flag.StringVar(&(dataString.EcsLaunchType), "ecsLaunchType", "", "EC2 or Fargate")
+	flag.StringVar(&(dataString.EcsDeploymentStrategy), "ecsDeploymentStrategy", "", "Daemon/Replica/Sidecar")
+	flag.StringVar(&(dataString.EcsClusterArn), "clusterArn", "", "Used to restart ecs task to apply new agent config")
+	flag.StringVar(&(dataString.CwagentConfigSsmParamName), "cwagentConfigSsmParamName", "", "Used to set new cwa config")
+	flag.StringVar(&(dataString.EcsServiceName), "cwagentECSServiceName", "", "Used to restart ecs task to apply new agent config")
+}
+
+func fillComputeType(e *MetaData, data *MetaDataStrings) *MetaData {
+	computeType, ok := compute_type.FromString(data.ComputeType)
+	if !ok {
+		panic("Invalid compute type. Needs to be EC2/ECS/EKS. Compute Type is a required flag. :" + data.ComputeType)
```

Comment:
**Nitpick**: We can use `log.panic` so it would have a better log and it also writes the logs to cloudwatch agent log too

Reply:
oh tyty [TODO]